### DISCUSSION TITLE
Best Practices for Backing Up Multiple Kubernetes Namespaces with Velero

### DISCUSSION THREAD

**Discussion by boedy** [vmware-tanzu/velero#7080] [General discussions]
Hello Velero Community,

I am currently fine-tuning my disaster recovery strategy using Velero and am faced with a decision that I'm sure many of you have tackled: when backing up a Kubernetes cluster with multiple namespaces, is it more advantageous to create a single scheduled backup for all namespaces, or to manage multiple scheduled backups, each including a single namespace?

Here's my context:
- My cluster has several namespaces, some of which are more critical than others.
- The amount of data and the rate of change vary across these namespaces.
- I am looking for a balance between operational simplicity and recovery effectiveness.

What is the recommended approach here?

**Comment by shawn-hurley** [comment]
Hello, 

This depends on the RPO and RTO that you have. If they significantly differ from namespace to namespace, I suggest grouping them based on RPO and RTO. 

One thing to note is that the backups happen sequentially, with only one backup at a time. So if you have a scheduled backup kick-off, say, five backups at the same time, they will only happen sometimes. 

I would also be curious what others think

**Comment by shubham-pampattiwar** [comment]
+1 to what @shawn-hurley is suggesting here. Additionally, 
- I would identify and prioritize critical namespaces, these may warrant individual backup schedules.
- For critical namespaces the backup schedules can be more frequent as compared to the non-critical ones.
- You could plan and execute a backup strategy right now and re-evaluate it down the line later and make changes accordingly.
- Overall I would suggest a hybrid approach.
